### ISSUE TITLE
System.Memory.Tests crashing on Windows outer loop runs

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#16492]
https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_windows_nt_debug/115/consoleText
https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_windows_nt_release/111/consoleText
https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_win10_debug/106/consoleText
https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_win10_release/112/consoleText
https://ci.dot.net/job/dotnet_corefx/job/master/job/code_coverage_windows/117/consoleText
```
  Discovering: System.Memory.Tests
  Discovered:  System.Memory.Tests
  Starting:    System.Memory.Tests
  Finished running tests.  End time= 3:21:30.59, Exit code = -1073741819
D:\j\workspace\outerloop_win---65ef7d78\Tools\tests.targets(249,5): warning MSB3073: The command "D:\j\workspace\outerloop_win---65ef7d78\bin/Windows_NT.AnyCPU.Release/System.Memory.Tests/netcoreapp//RunTests.cmd D:\j\workspace\outerloop_win---65ef7d78\bin/testhost/netcoreapp-Windows_NT-Release-x64/" exited with code -1073741819. [D:\j\workspace\outerloop_win---65ef7d78\src\System.Memory\tests\System.Memory.Tests.csproj]
D:\j\workspace\outerloop_win---65ef7d78\Tools\tests.targets(257,5): error : One or more tests failed while running tests from 'System.Memory.Tests' please check D:\j\workspace\outerloop_win---65ef7d78\bin/Windows_NT.AnyCPU.Release/System.Memory.Tests/netcoreapp/testResults.xml for details! [D:\j\workspace\outerloop_win---65ef7d78\src\System.Memory\tests\System.Memory.Tests.csproj]
  Using D:\j\workspace\outerloop_win---65ef7d78\bin\testhost\netcoreapp-Windows_NT-Release-x64\ as the test runtime folder.
```
cc: @jkotas

**Comment by jkotas** [comment]
@ahsonkhan This is regression from your Span.Clear optimization. It is crashing at:

```
coreclr!memset+0x49 [f:\dd\vctools\crt\vcruntime\src\string\amd64\memset.asm @ 80]
System_Private_CoreLib_ni!System.Runtime.RuntimeImports.RhZeroMemory(Byte ByRef, UInt64)+0x76 [E:\A\_work\127\s\src\mscorlib\src\System\Runtime\RuntimeImports.cs @ 25]
System_Private_CoreLib_ni!System.Span`1[[System.Int32, System.Private.CoreLib]].Clear()+0x18 [E:\A\_work\127\s\src\mscorlib\src\System\Span.cs @ 253]
System_Memory_Tests!System.SpanTests.SpanTests.ClearLongerThanUintMaxValueBytes()+0xe9 [C:\corefx\src\System.Memory\tests\Span\Clear.cs @ 227]
```

**Comment by shiftylogic** [comment]
I've tested this method with variations using up to 8 GB of memory on both 32-bit (only up to 2 GB here) and 64-bit without reproducing this bug.

Either a change since the bug was discovered has fixed it, or something else triggers this that my sample app is not doing.

**Comment by shiftylogic** [comment]
I've made a tweak to these tests to have the 2 large allocation tests run serially to each other. I made an initial PR (#17441) that moved them to the inner loop and ran the CI system a few times.

There was also some changes made in the clear implementation a bit ago that may have also resolved these issues.

We will monitor the outerloop builds and see if the issue returns (disabling the tests if necessary).
